### PR TITLE
Update syntax highlighting for Emacs

### PR DESCRIPTION
**Pull request by Piturnah**
Since the most recent stream was dedicated to bikeshedding the syntax, I've updated the noq-mode.el to reflect the changes!

image
image

Hope you like it!

### CODE REVIEW

**Comment by Piturnah** [comment]
Note the comment on line 57:
```elisp
    ;; FIXME: It would be cool if variables were highlighted too (see below). However, there is an
    ;; issue in that a number can either be a symbol or a variable depending on the context.
    ;; The regex below works around this by only highlighting variables that don't start with a
    ;; number, however this obviously won't cover all possible variables.
```
In the end, I left the regex rule uncommented, as looking through all of the provided examples and standard library there wasn't any cases where this applied. However, it's still worth noting, and worth considering removing, in my opinion

**Comment by rexim** [comment]
> However, there is an issue in that a number can either be a symbol or a variable depending on the context

No, number is always a symbol. Variable is any identifier that starts with a capital letter or an underscore `_`: https://github.com/tsoding/Noq/blob/62d9e6d4408524657abe3bf1e27e8eeeb22722a6/src/main.rs#L196-L203

File: editor/noq-mode.el [hunk 1]
```diff
@@ -43,11 +43,26 @@
   "Syntax table for `noq-mode'.")
 
 (eval-and-compile
-  (defconst noq-keywords
-    '("rule" "shape" "apply" "done" "quit" "undo" "reverse" "load")))
```
Comment (rexim) [inline editor/noq-mode.el:47]: `undo`, `quit`, `delete` and `load` are still actively used keywords:

https://github.com/tsoding/Noq/blob/62d9e6d4408524657abe3bf1e27e8eeeb22722a6/src/lexer.rs#L35-L39

**Review by rexim — CHANGES_REQUESTED** [review]

**Comment by rexim** [comment]
I don't know why but variables are not highlighted for me:

2022-04-12-201929_912x858_scrot

Commit: 287e5316dfe1f0baafcfa548c5adb9bab6e8c94d
Version: GNU Emacs 27.1

**Comment by Piturnah** [comment]
I suspect this is because gruber-darker doesn't define highlighting for variables, you can see with `M-x list-faces-display` under `font-lock-variable-name-face`. It shows on most other themes, I changed to this face in commit 701eaed since I thought it would match the semantic better

Before it was using `font-lock-builtin-face`

**Comment by rexim** [comment]
@Piturnah oh I see, it uses a color that is almost indistinguishable from the foreground: 

https://github.com/rexim/gruber-darker-theme/blob/72278089c440d45c00fb8afcd53af82fd30f451b/gruber-darker-theme.el#L186

Thanks for the explanation! And thanks for the contribution!

**Review by rexim — APPROVED** [review]

**Pull request merged**